### PR TITLE
supplemental: Issue#14273 Add JUnit test coverage for overloaded private and static methods for OverloadMethodsDeclarationOrder

### PR DESCRIPTION
**Pull request by Zopsss**
Issue: #14273 

> Thanks a lot for testing modifiers, please send PR to contribute Junit test coverage to make sure it will always be like this:).

Added JUnit Test coverage for private and static methods as a part of `OverloadMethodsDeclarationOrder` check.

Created a new input file called `InputOverloadMethodsDeclarationOrderPrivateAndStaticMethods.java` and wrote the test method for it.

### CODE REVIEW

**Comment by Zopsss** [comment]
Hey @romani can you help me with the failing CI check? I'm not able to understand it.l

**Comment by romani** [comment]
please pull latest master from our repo, we disabled sonar yesterday

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java [hunk 1]
```diff
@@ -60,6 +61,17 @@ public void testOverloadMethodsDeclarationOrderRecords() throws Exception {
             expected);
     }
 
+    @Test
+    public void testOverloadMethodsDeclarationOrderPrivateAndStaticMethods() throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java:65]: in addition please add this test to
https://github.com/checkstyle/checkstyle/blob/master/src/it/java/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/OverloadMethodsDeclarationOrderTest.java

to make it firm that no matter how this Check might change in design, google style config works good on such code snippet.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java [hunk 2]
```diff
@@ -60,6 +61,17 @@ public void testOverloadMethodsDeclarationOrderRecords() throws Exception {
...
+
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
```
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java:70]: It was giving some kind of error message so I had to use the non compile-able way. Should I add an comment mentioning that it should be compiled with jdk11 before the class?

[the remark this answers is not in the document]
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java]: sorry, you were correct. The test started passing after I put the code to compiled folder and then ran the test with `getFile()`. I have made the changes as you requested and pushed it.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java (hunk 1 above)
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java:65]: done

**Review by romani — APPROVED** [review]
Thanks a lot, ok to merge

**Pull request merged**